app/src/main/java/org/jellyfin/androidtv/ui/playback/nextup/NextUpViewModel.kt

```diff
@@ -54,20 +60,42 @@ class NextUpViewModel(
 	private suspend fun loadItemData(id: String) = withContext(Dispatchers.IO) {
 		val item = apiClient.getItem(id) ?: return@withContext null
 
-		val thumbnail = apiClient.GetImageUrl(item, ImageOptions())
+		val thumbnail = when (userPreferences[UserPreferences.nextUpBehavior]) {
+			NextUpBehavior.EXTENDED -> apiClient.GetImageUrl(item, ImageOptions())
+			else -> null
+		}
 		val logo = apiClient.GetLogoImageUrl(item, ImageOptions())
 
-		val title = if (item.indexNumber != null && item.name != null)
-			"${item.indexNumber}. ${item.name}"
-		else if (item.name != null)
-			item.name
-		else ""
+		val seasonNumber = when (item.baseItemType == BaseItemType.Episode && item.parentIndexNumber != null
+			&& item.parentIndexNumber != 0) {
+			true -> context.getString(R.string.lbl_season_number, item.parentIndexNumber)
+			false -> null
+		}
+		val episodeNumber = when {
+			(item.baseItemType != BaseItemType.Episode) ->
+				item.indexNumber?.toString()
+			(item.parentIndexNumber == 0) ->
+				context.getString(R.string.lbl_special)
+			(item.indexNumber != null && item.indexNumberEnd != null) ->
+				context.getString(R.string.lbl_episode_range, item.indexNumber, item.indexNumberEnd)
+			(item.indexNumber != null) ->
+				context.getString(R.string.lbl_episode_number, item.indexNumber)
+			else -> null
+		}
```

Comment:
```suggestion
		val episodeNumber = when {
			item.baseItemType != BaseItemType.Episode -> item.indexNumber?.toString()
			item.parentIndexNumber == 0 -> context.getString(R.string.lbl_special)
			item.indexNumber != null -> when {
				item.indexNumberEnd != null -> context.getString(R.string.lbl_episode_range, item.indexNumber, item.indexNumberEnd)
				else -> context.getString(R.string.lbl_episode_number, item.indexNumber)
			}
			else -> null
		}
```
Same code but with the values on the same line and a secondary when statement for the indexnumberend/no end